### PR TITLE
🚧  First draft

### PR DESCRIPTION
**Pull request by frinyvonnick**
- [x] Add tests
- [x] Update README.md
~~Add examples~~

### CODE REVIEW

File: src/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    equals(wrapper, SecondComponent)
+  })
+
+  it(`should throw an error when prop value is undefined and a default
+    component isn't provided`, () => {
+    try {
+      getWrapped({
+        states: {
+          0: FirstComponent,
+          1: SecondComponent,
```
Comment (nlepage) [inline src/index.spec.js:89]: Add a :
```
undefined: ThridComponent,
```
to check that `undefined` is not coerced in string

**Review by nlepage — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "hoc-react-switcher",
+  "version": "0.1.0",
+  "description": "Display a components depending on prop value",
```
Comment (fabienjuif) [inline package.json]: Ask @hgwood, I dunno if this is a good english sentence

File: src/index.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+
+export const errorMessage = `Your prop value should match one of the \`states\` or
```
Comment (fabienjuif) [inline src/index.jsx]: Why do you use a ` ?
This should be enough : 
```es6
'Your prop value should match one of the `states` or you should provide a `DefaultComponent`'
```

File: src/index.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const {
+    prop = 'state',
+    states = {},
+    DefaultComponent,
+  } = config || {}
+
+  return (props) => {
+    const propValue = props[prop]
+
+    if (propValue === undefined && !DefaultComponent) {
```
Comment (fabienjuif) [inline src/index.jsx]: Why this condition is not a one-liner and the next one is ?

File: package.json (hunk 2 above)
Comment (hgwood) [reply package.json]: What about "React component which displays one of its children depending on a prop value"?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Parameter | Required | Default value | Description
+----------|--------|---------------|-------------
+`prop` | no | `state` | The prop to watch. It determines which component to display
+`states` | yes | `{}` | A map of components
+`DefaultComponent` | no | `undefined` | A component to display when prop value doesn't match one of the states
+
+
+**Component.js**
+```(javascript)
```
Comment (mbernardeau) [inline README.md]: You can specify JSX language for better syntax highlighting:

```JSX
import React, { PropTypes } from 'react'
import switcher from 'hoc-react-switcher'

const FirstComponent = () => {
  return (
    <div>
      FirstComponent
    </div>
  )
}
```

**Pull request merged**